src/main.cpp

```diff
@@ -6535,7 +6535,7 @@ bool static AlreadyHave(const CInv& inv) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
     return true;
 }
 
-void static ProcessGetData(CNode* pfrom)
+void static ProcessGetData(CNode* pfrom, std::atomic<bool>& interruptMsgProc)
```

Comment:
Just a small thing here, I would suggest making `interruptMsgProc` a const ref, to make it clear that it is read only from this function.
Same thing for all the other functions that take a reference to `interruptMsgProc` in `main.cpp`.

Reply:
Thanks! I agree with your comment. Code changes have been updated as suggested.